it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/overseasControllers/CaptureOverseasTaxIdentifierControllerISpec.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers.overseasControllers
+
+import play.api.libs.ws.WSResponse
+import play.api.test.Helpers.{BAD_REQUEST, NO_CONTENT, OK, SEE_OTHER, await, defaultAwaitTimeout}
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.controllers.overseasControllers
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, StorageStub}
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ComponentSpecHelper
+import uk.gov.hmrc.minorentityidentificationfrontend.views.overseasViews.CaptureOverseasTaxIdentifierTests
+
+
+class CaptureOverseasTaxIdentifierControllerISpec extends ComponentSpecHelper
+  with AuthStub
+  with StorageStub
+  with CaptureOverseasTaxIdentifierTests {
+
+  "GET /overseas-tax-identifier" when {
+
+    "displayed" should {
+
+      lazy val result = {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          internalId = testInternalId,
+          testOverseasCompanyJourneyConfig(businessVerificationCheck = true)
+        ))
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        get(s"/identify-your-overseas-business/$testJourneyId/overseas-tax-identifier")
+      }
+
+      "return OK" in {
+        result.status mustBe OK
+      }
+
+      "return a view which" should {
+        testOverseasTaxIdentifierCommonViewTests(result)
+        testInitialCaptureOverseasTaxIdentifierView(result)
+      }
+
+    }
+
+    "an unauthorized user attempts to use the service" should {
+
+      "redirect the user to the sign on page" in {
+        stubAuthFailure()
+        lazy val result: WSResponse = get(s"/identify-your-overseas-business/$testJourneyId/overseas-tax-identifier")
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri("/bas-gateway/sign-in" +
+            s"?continue_url=%2Fidentify-your-overseas-business%2F$testJourneyId%2Foverseas-tax-identifier" +
+            "&origin=minor-entity-identification-frontend"
+          )
+        )
+      }
+    }
+
+  }
+
+  "POST /overseas-tax-identifier" when {
+
+    "the user selects 'Yes' and the tax identifier is correctly formatted" should {
+
+      "redirect to Check Your Answers" in {
```

Comment:
Believe it goes to OTI country page which will be done in separate ticket. Would be nice put a comment something

Reply:
Done